### PR TITLE
Stricter CORS headers

### PR DESCRIPTION
**Pull request by atosatto**
Trying to fix issue #56 during the Torino HackNight contest.

The idea is basically to select a pair of free port from port-range using the `net.Listen` function.
The two ports are later used by gin to listen for incoming HTTP or HTTPS connections.

### CODE REVIEW

File: main.go [hunk 1]
```diff
@@ -215,11 +232,18 @@ func main() {
...
 
-			extraOriginStr := "https://create.arduino.cc, http://create.arduino.cc, https://create-dev.arduino.cc, http://create-dev.arduino.cc, http://create-staging.arduino.cc, https://create-staging.arduino.cc"
-
-			for i := 8990; i < 9001; i++ {
-				extraOriginStr = extraOriginStr + ", http://localhost:" + strconv.Itoa(i) + ", https://localhost:" + strconv.Itoa(i)
+			portPlain, err := getBindPort(8990, 9001)
+			if err != nil {
+				panic(err)
 			}
+			portSSL, err := getBindPort(portPlain+1, 9001)
```
Comment (matteosuppo) [inline main.go]: You can maybe add a comment here to explain why you're passing portPlain+1. I heard people confused by it in the office. But of course if you find a free port, the previous are all occupied, and there's no reason to cycle through them again.

**Comment by matteosuppo** [comment]
I tested it with different listening services on different ports and it works. Just add that comment and we can merge it.

**Comment by atosatto** [comment]
I've added a comment as you suggested. Maybe it's a little bit verbose, but it should clarify why I'm passing `portPlain+1`.

**Comment by matteosuppo** [comment]
Thank you very much

**Pull request merged**